internal/server/mcp.go

```diff
@@ -71,6 +84,30 @@ func (m *sseManager) remove(id string) {
 	m.mu.Unlock()
 }
 
+func (m *sseManager) cleanupRoutine(ctx context.Context) {
+	timeout := 10 * time.Minute
+	ticker := time.NewTicker(timeout)
+	defer ticker.Stop()
+
+	for range ticker.C {
+		m.mu.Lock()
+		now := time.Now()
+		for id, sess := range m.sseSessions {
+			if now.Sub(sess.lastActive) > timeout {
+				delete(m.sseSessions, id)
+			}
+		}
+		m.mu.Unlock()
```

Comment:
nit: pro-tip for these situations is you can still use defer in an anon function to ensure the mutex is unlocked even if a panic occurs
```suggestion
                 // use a function here in order to user 'defer' to unlock this mutex when complete
		func() {
		  m.mu.Lock()
		  defer m.mu.Unlock()
		  now := time.Now()
		  for id, sess := range m.sseSessions {
			  if now.Sub(sess.lastActive) > timeout {
				  delete(m.sseSessions, id)
			  }
		  }
		 }()
		
```